src/Discovery/DiscoveryDetails/DataDownloadList/DataDownloadList.tsx

```diff
@@ -1,28 +1,53 @@
 import React from 'react';
 import { List } from 'antd';
-import DataDownloadListItem from './Utils/DataDownloadListItem';
+import DataDownloadListItem from './Interfaces/DataDownloadListItem';
 import './DataDownloadList.css';
 import ProcessData from './Utils/ProcessData';
+import ActionButtons from './ActionButtons/ActionButtons';
+import { DiscoveryConfig } from '../../DiscoveryConfig';
+import { DiscoveryResource } from '../../Discovery';
 
-const DataDownloadList = ({ sourceFieldData }: any) => {
+interface DataDownloadListProps {
+  isUserLoggedIn: boolean;
+  discoveryConfig: DiscoveryConfig;
+  resourceInfo: DiscoveryResource;
+  sourceFieldData: any[];
+  healLoginNeeded: boolean;
+}
+
+const DataDownloadList = ({
+  isUserLoggedIn,
+  discoveryConfig,
+  resourceInfo,
+  sourceFieldData,
+  healLoginNeeded,
+}: DataDownloadListProps) => {
   const data = ProcessData(sourceFieldData);
-  if (data.length === 0) {
-    return null;
-  }
+  const noData = data.length === 0;
```

Comment:
I think this is really hard to show up if we are on the discovery details page, because this basically means the selected study has an empty `__manifest` field. But in this case, the `DataDownloadList` component won't be showing up in the first place due to https://github.com/uc-cdis/data-portal/blob/bd988afb03b2271a000d44cb5117ccf7a12554d5/src/Discovery/DiscoveryDetails/DiscoveryDetails.tsx#L261-L266
For now maybe this is an acceptable behavior, since these buttons are part of the `DataDownloadList` component, make only part of them to show up is kind of tricky. And we can defer them to a later ticket. But can you update to code to add some comments in there to say why this popover won't show up and how can we fix it later?

Reply:
Instead of adding code comments that might end up living in the code for a long time and end up out of date, could we instead create a ticket in JIRA that explains the requirement and identifies in the technical details this section of code that needs to be changed? I've found this approach helps avoid code bases with comments that are no longer relevant, difficult to understand and situations where developers are unsure if they can or should delete the comments. Let me know what you think; I can add the code comments if you prefer.

Reply:
sounds good, i can create a ticket